core/src/main/java/com/linecorp/armeria/server/HttpHeaderPathMapping.java

```diff
@@ -80,11 +80,13 @@
     public PathMappingResult apply(PathMappingContext mappingCtx) {
         final PathMappingResult result = pathStringMapping.apply(mappingCtx);
         if (!result.isPresent()) {
+            if (mappingCtx.isPreflight() && !mappingCtx.delayedThrowable().isPresent()) {
+                mappingCtx.delayThrowable(HttpStatusException.of(HttpStatus.FORBIDDEN));
+            }
             return PathMappingResult.empty();
         }
 
-        // We need to check the method at the last in order to return '405 Method Not Allowed'.
-        if (!supportedMethods.contains(mappingCtx.method())) {
+        if (!mappingCtx.isPreflight() && !supportedMethods.contains(mappingCtx.method())) {
```

Comment:
How about updating the comment instead of just removing it?

Reply:
Got it. Will update it. :-)